### PR TITLE
Javascript/javascript3/week1

### PR DESCRIPTION
**Pull request by SlaviaF**

### CODE REVIEW

File: javascript/javascript3/week1/giphy.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+//create img and append it the document
+
+const searchGifBtn = document.getElementById("searchGifBtn")
+const gifType = document.getElementById("gifType")
+const noOfGif = document.getElementById("noOfGif")
+
+let searchString;
+let gifNum;
+searchGifBtn.addEventListener("click", getGiphy)
+function getGiphy() {
```
Comment (senner007) [inline javascript/javascript3/week1/giphy.js:13]: Think about how you can get your functions to do one thing only. And how this action must be evident just be reading the name. Usually this means writing small functions. Its called "The single responsibility principle"

Comment (senner007) [inline javascript/javascript3/week1/giphy.js]: Think about how you can get your functions to do one thing only. And how this action must be evident just be reading the name. Usually this means writing small functions. Its called "The single responsibility principle"

File: javascript/javascript3/week1/weatherApp.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        .then(response => response.json())
+        .then(weatherData => {
+            renderWeatherStatus(weatherData)
+            localStorage.setItem("userLocation", renderWeatherStatus(weatherData))
+
+        })
+}
+//Get weather from  user location
+currentLctBtn.addEventListener('click', userLocation);
+function userLocation() {
```
Comment (senner007) [inline javascript/javascript3/week1/weatherApp.js]: Make sure the name of the function always contains atleast one verb

**Review by senner007 — APPROVED** [review]
Good !
You seem to have a good understanding of this.
A few minor comments :
Make sure to think about error handling also. Think about what happens when users do unexpected things. 
What happens ehen you enter a misspelled city in te weather app for instance?
The giphy app also needs to update according to the number of images chosen. 

All in all very good! ✔

File: javascript/javascript3/week1/giphy.js (hunk 1 above)
Comment (SlaviaF) [reply javascript/javascript3/week1/giphy.js]: Yes ...Got it . Will make sure to follow the principle hence forth

File: javascript/javascript3/week1/weatherApp.js (hunk 2 above)
Comment (SlaviaF) [reply javascript/javascript3/week1/weatherApp.js]: Have changed it now and will careful the next time

**Comment by SlaviaF** [comment]
> Good !
> You seem to have a good understanding of this.
> A few minor comments :
> Make sure to think about error handling also. Think about what happens when users do unexpected things.
> What happens ehen you enter a misspelled city in te weather app for instance?
> The giphy app also needs to update according to the number of images chosen.
> 
> All in all very good! ✔

Hi Niels, 
I now have some error handling in the weather app and I realised I have become more comfortable with error handling after learning about promises and error handling. I have also updated the images in the Giphy app.

**Pull request merged**